src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -449,6 +449,13 @@ private StrongNameKeys ComputeStrongNameKeys()
             // when both attributes and command-line options specified, cmd line wins.
             string keyFile = _compilation.Options.CryptoKeyFile;
 
+            // Public sign requires a keyfile
+            if (DeclaringCompilation.Options.PublicSign)
+            {
+                // If we're public signing, we don't need a strong name provider
+                return StrongNameKeys.Create(keyFile, MessageProvider.Instance);
```

Comment:
+1 for the last option: reporting error that the attribute shouldn't be used at all.